src/client.js

```diff
@@ -51,35 +51,40 @@ export default class DockerExecWebsocketClient extends EventEmitter {
     //set state, state does nothing yet
     this.state = msgcode.pause;
 
-    this.stdin = through((data) => {
+    this.stdin = through2((data, enc, cb) => {
       this.sendMessage(msgcode.stdin, data);
-    });
-
-    this.stdin.on('end', () => {
+      cb();
+    }, (cb) => {
       this.sendCode(msgcode.end);
+      cb();
     });
 
-    //stream with pause buffering, everything passes through here first
-    this.strbuf = through();
+    //stream with pause buffering, everything passes thru here first
+    this.strbuf = through2();
 
     const MAX_OUTSTANDING_BYTES = 8 * 1024 * 1024;
     this.outstandingBytes = 0;
 
-    this.strbuf.pause();
-    this.strbuf.pipe(through((data) => {
+    this.strbuf.pipe(through2((data, enc, cb) => {
       this.outstandingBytes += data.length;
       this.socket.send(data, {binary: true}, () => {
         this.outstandingBytes -= data.length;
       });
       if (this.outstandingBytes > MAX_OUTSTANDING_BYTES) {
+        this.state = msgcode.pause;
         this.strbuf.pause();
+        this.emit('paused');
       } else {
+        this.state = msgcode.resume;
         this.strbuf.resume();
+        this.emit('resumed');
       }
+      cb();
     }));
+    this.strbuf.pause();
```

Comment:
So even if user of this library don't want to read `stdout` or `stderr` the user still have to call `client.stderr.resume()` and  `client.stderr.resume()`... But you don't have to listen for the `.on('data'...)`...